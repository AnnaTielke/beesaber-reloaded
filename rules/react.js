module.exports = {
  plugins: ['react'],

  extends: ['plugin:react/recommended'],

  settings: {
    react: {
      version: 'detect',
    },
  },

  rules: {
    'react/button-has-type': 'error',
    'react/jsx-boolean-value': 'error',
    'react/jsx-fragments': 'error',
    'react/jsx-handler-names': 'error',
    'react/jsx-no-bind': ['error', { allowArrowFunctions: true }],
    'react/jsx-pascal-case': 'error',
    'react/no-access-state-in-setstate': 'error',
    'react/no-array-index-key': 'error',
    'react/no-danger': 'error',
    'react/no-did-mount-set-state': 'error',
    'react/no-did-update-set-state': 'error',
    'react/no-multi-comp': 'error',
    'react/no-this-in-sfc': 'error',
    'react/no-typos': 'error',
    'react/no-unsafe': 'error',
    'react/no-unused-prop-types': 'error',
    'react/no-unused-state': 'error',
    'react/no-will-update-set-state': 'error',
    'react/prefer-es6-class': 'error',
    'react/prefer-stateless-function': 'error',
    'react/style-prop-object': 'error',
    'react/void-dom-elements-no-children': 'error',
  },
};
