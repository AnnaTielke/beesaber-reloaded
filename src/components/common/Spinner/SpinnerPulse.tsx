import styled from 'styled-components';

import tokens from '@/tokens/index';

const Container = styled.div`
  @keyframes ldio-do9usc7wgsf {
    0% {
      top: 90px;
      left: 90px;
      width: 0;
      height: 0;
      opacity: 1;
    }
    100% {
      top: 15px;
      left: 15px;
      width: 150px;
      height: 150px;
      opacity: 0;
    }
  }
  .ldio-do9usc7wgsf div {
    position: absolute;
    border-width: 10px;
    border-style: solid;
    opacity: 1;
    border-radius: 50%;
    animation: ldio-do9usc7wgsf 1s cubic-bezier(0, 0.2, 0.8, 1) infinite;
  }
  .ldio-do9usc7wgsf div:nth-child(1) {
    border-color: ${tokens.color.yellow.main};
    animation-delay: 0s;
  }
  .ldio-do9usc7wgsf div:nth-child(2) {
    border-color: ${tokens.color.blue.main};
    animation-delay: -0.5s;
  }
  .loadingio-spinner-ripple-rt8n616prbh {
    width: 200px;
    height: 200px;
    display: inline-block;
    overflow: hidden;
    background: ${tokens.color.page.bgColor};
  }
  .ldio-do9usc7wgsf {
    width: 100%;
    height: 100%;
    position: relative;
    transform: translateZ(0) scale(1);
    backface-visibility: hidden;
    transform-origin: 0 0; /* see note above */
  }
  .ldio-do9usc7wgsf div {
    box-sizing: content-box;
  }
`;

function SpinnerPulse(): JSX.Element {
  return (
    <Container>
      <div className="loadingio-spinner-ripple-rt8n616prbh">
        <div className="ldio-do9usc7wgsf">
          <div></div>
          <div></div>
        </div>
      </div>
    </Container>
  );
}

export default SpinnerPulse;
