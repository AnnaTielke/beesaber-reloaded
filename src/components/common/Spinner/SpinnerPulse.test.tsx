import { render } from '@testing-library/react';

import SpinnerPulse from './SpinnerPulse';

describe('SpinnerPulse', () => {
  it('should match the snapshot', () => {
    const { container } = render(<SpinnerPulse />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
