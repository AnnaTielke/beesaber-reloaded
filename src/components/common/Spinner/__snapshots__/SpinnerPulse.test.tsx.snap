// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpinnerPulse should match the snapshot 1`] = `
.c0 .ldio-do9usc7wgsf div {
  position: absolute;
  border-width: 10px;
  border-style: solid;
  opacity: 1;
  border-radius: 50%;
  -webkit-animation: ldio-do9usc7wgsf 1s cubic-bezier(0,0.2,0.8,1) infinite;
  animation: ldio-do9usc7wgsf 1s cubic-bezier(0,0.2,0.8,1) infinite;
}

.c0 .ldio-do9usc7wgsf div:nth-child(1) {
  border-color: #ffeb00;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
}

.c0 .ldio-do9usc7wgsf div:nth-child(2) {
  border-color: #0000FF;
  -webkit-animation-delay: -0.5s;
  animation-delay: -0.5s;
}

.c0 .loadingio-spinner-ripple-rt8n616prbh {
  width: 200px;
  height: 200px;
  display: inline-block;
  overflow: hidden;
  background: #00111f;
}

.c0 .ldio-do9usc7wgsf {
  width: 100%;
  height: 100%;
  position: relative;
  -webkit-transform: translateZ(0) scale(1);
  -ms-transform: translateZ(0) scale(1);
  transform: translateZ(0) scale(1);
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
  -webkit-transform-origin: 0 0;
  -ms-transform-origin: 0 0;
  transform-origin: 0 0;
}

.c0 .ldio-do9usc7wgsf div {
  box-sizing: content-box;
}

<div
  class="c0"
>
  <div
    class="loadingio-spinner-ripple-rt8n616prbh"
  >
    <div
      class="ldio-do9usc7wgsf"
    >
      <div />
      <div />
    </div>
  </div>
</div>
`;
